src/test/java/liuni/TwitterStatusTests.java

```
@@ -0,0 +1,93 @@
+        try {
+            when(twitter.updateStatus(testString)).thenReturn(status);
+            when(status.getText()).thenReturn(testString);
+
+            testResult = postTestString(testString);
+        }
+        catch (Exception e) {
+            testResult = false;
+        }
+        assertTrue(testResult);
+        reset(twitter);
+        reset(status);
+    }
+
+    @Test
+    public void testTweetPost_EmptyTweet() {
+        String testString = "";
+        boolean testResult;
+        try {
+            when(twitter.updateStatus(testString)).thenReturn(status);
+            when(status.getText()).thenReturn(testString);
+
+            testResult = postTestString(testString);
+        }
+        catch (Exception e) {
+            testResult = false;
+        }
+        assertFalse(testResult);
+        reset(twitter);
+        reset(status);
+    }
+
+    @Test
+    public void testTweetPost_TooLongTweet() {
+        String testString = "This is a bad string because it is going go to be longer than Twitter's limits. It is this long because it is Lorem Ipsum. Lorem ipsum dolor sit amet, consectetur adipiscing elit. In vehicula velit vel augue porta condimentum. Nullam ornare velit mattis, maximus sem eget, malesuada nibh. Interdum et malesuada fames ac ante ipsum primis in faucibus. Suspendisse potenti. Pellentesque a dictum lorem. Ut sit amet fringilla turpis. Sed pretium, erat tincidunt aliquam mattis, neque diam luctus mauris, sed lacinia arcu libero sit amet tortor.";
+        boolean testResult;
+        try {
+            when(twitter.updateStatus(testString)).thenReturn(status);
+            when(status.getText()).thenReturn(testString);
+
+            testResult = postTestString(testString);
+        }
+        catch (Exception e) {
+            testResult = false;
+        }
+        assertFalse(testResult);
+        reset(twitter);
+        reset(status);
+    }
+
+    private boolean postTestString(String testString) {
```

while helper methods are nice, you're also masking the Exception and false... you could end up with bad unit tests

---

I may need this explained to me...

---

Nevermind, think I understand.